### PR TITLE
Add cobra cli

### LINKED ISSUE #5: Improve CLI UX

### PR DESCRIPTION
**Pull request by Benco1**
- Use cobra to create CLI with commands, flags, etc... according to linked issue.

- Repurposed former `main` as a shared function `benchmark` (see `cmd/shared.go`). I imagine contents will eventually be refactored, but didn't want to convolve the two issues. Included this function in the `cmd` package, particularly to share with `localCmd` and `remoteCmd`.

- Moved `dnsbench.go` to it's own directory to resolve some import issues. (Perhaps there is a better solution to this?)

- As we discussed in Slack, the `version` command is currently stubbed to print `master` as version and holding off on the `transport` flag for now.

Closes #5.

### CODE REVIEW

File: pkg/cmd/shared.go [hunk 1]
```diff
@@ -1,41 +1,20 @@
...
 
-	"github.com/askcom/dnsbench"
+	"github.com/askcom/dnsbench/pkg/resolve"
 	"github.com/codahale/hdrhistogram"
 )
 
-func main() {
-	var nameserver string
+// benchmark is a shared function to execute benchmarking in either local or remote context
+func benchmark(nameserver string, localResolver *bool, concur *int, count *int, interval *time.Duration, namesfile *string, qps *int) {
```
Comment (richardmarshall) [inline pkg/cmd/shared.go]: What is the intent behind all these arguments being pointers?

File: pkg/cmd/version.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+	"github.com/spf13/cobra"
+)
+
+// versionCmd represents the version command
+var versionCmd = &cobra.Command{
+	Use:   "version",
+	Short: "Display program version.",
+	Run: func(cmd *cobra.Command, args []string) {
+		var version = "master"
```
Comment (richardmarshall) [inline pkg/cmd/version.go]: This should be a package level global to enable us to set it via linker flags.

File: pkg/cmd/shared.go (hunk 1 above)
Comment (Benco1) [reply pkg/cmd/shared.go]: I believe they were pointers in the initial main function and wanted to minimize changes to the original code at this stage.

Comment (richardmarshall) [reply pkg/cmd/shared.go]: Got it, so you didn't need to remove the dereferences in the code inside this new function. That's fine.

**Comment by richardmarshall** [comment]
LGTM. Onward with the continued refactoring!

**Pull request merged**